dipy/reconst/peak_direction_getter.pyx

```diff
@@ -75,8 +75,12 @@ cdef class PeaksAndMetricsDirectionGetter(DirectionGetter):
                 raise IndexError("point outside data")
 
         # Check to see how many peaks were found in the voxel
-        for numpeaks in range(self._ind.shape[3]):
-            if self._ind[ijk[0], ijk[1], ijk[2], numpeaks] < 0:
+        numpeaks=0
+        for i in range(self.peak_values.shape[3]):
+            # Test if the value associated to the peak is > 0
+            if self._qa[ijk[0], ijk[1], ijk[2], i] > 0:
+                numpeaks = numpeaks + 1
+            else:
```

Comment:
Thanks! This is certainly clearer. 

Would:

```
numpeaks = sum(self._qa[ijk[0], ijk[1], ijk[2], i] > 0) 
```

work here?

Reply:
No, `>` doesn't work for arrays. Am I missing something?
```
Error compiling Cython file:
------------------------------------------------------------
...
         numpeaks = sum(self._qa[ijk[0], ijk[1], ijk[2], :] > 0)
                                                          ^
------------------------------------------------------------

dipy/reconst/peak_direction_getter.pyx:85:59: Invalid types for '>' (double[::1], long)

```

Would have been more clear :)

Reply:
Oh well. Thanks for checking.